src/gui/guiFormSpecMenu.cpp

```diff
@@ -1731,25 +1731,27 @@ void GUIFormSpecMenu::parseLabel(parserData* data, const std::string &element)
 		return;
 
 	std::vector<std::string> v_pos = split(parts[0],',');
-	std::string text = parts[1];
 
 	MY_CHECKPOS("label",0);
 
 	if(!data->explicit_size)
 		warningstream<<"invalid use of label without a size[] element"<<std::endl;
 
-	std::vector<std::string> lines = split(text, '\n');
-
 	auto style = getDefaultStyleForElement("label", "");
 	gui::IGUIFont *font = style.getFont();
 	if (!font)
 		font = m_font;
 
-	for (unsigned int i = 0; i != lines.size(); i++) {
-		std::wstring wlabel_colors = translate_string(
-			utf8_to_wide(unescape_string(lines[i])));
-		// Without color escapes to get the font dimensions
-		std::wstring wlabel_plain = unescape_enriched(wlabel_colors);
+	EnrichedString str(unescape_string(utf8_to_wide(parts[1])));
+	size_t str_pos = 0;
+
+	for (size_t i = 0; str_pos < str.size(); ++i) {
+		// Split per line
+		size_t str_nl = str.getString().find(L'\n', str_pos);
+		if (str_nl == std::wstring::npos)
+			str_nl = str.getString().size();
+		EnrichedString line(str.substr(str_pos, str_nl - str_pos));
```

Comment:
this is our custom substr method, right?
could it directly return an EnrichedString?

Reply:
It already returns an enriched string.

Reply:
weird to invoke the constructor then, what about this?
```suggestion
		EnrichedString line = str.substr(str_pos, str_nl - str_pos);
```

Reply:
Right. I tend to use constructors over potential constructor + assign operators.